source/IntegrationTests/Infrastructure/Migration/JsonTransformerTests.cs

```diff
@@ -0,0 +1,283 @@
+using Energinet.DataHub.EDI.OutgoingMessages.Domain.DocumentWriters.RSM012;
+using Energinet.DataHub.EDI.OutgoingMessages.Domain.Models.OutgoingMessages;
+using Energinet.DataHub.EDI.OutgoingMessages.IntegrationTests.DocumentAsserters;
+using Energinet.DataHub.EDI.OutgoingMessages.UnitTests.Domain.RSM012;
+using NodaTime.Extensions;
+using NodaTime.Text;
+using Xunit;
+using Period = Energinet.DataHub.EDI.BuildingBlocks.Domain.Models.Period;
+
+namespace Energinet.DataHub.EDI.IntegrationTests.Infrastructure.Migration;
+
+public class JsonTransformerTests
+{
+    private static readonly string TestJson = """
+    {
+    "MeteredDataTimeSeriesDH3": {
+    "Header": {
+    "MessageId": "13255042",
+    "DocumentType": "E66",
+    "Creation": "2024-01-16T07:55:33Z",
+    "EnergyBusinessProcess": "D42",
+    "EnergyIndustryClassification": "23",
+    "SenderIdentification": {
+    "SchemeAgencyIdentifier": "9",
+    "content": "5790001330552"
+    },
+    "RecipientIdentification": {
+    "SchemeAgencyIdentifier": "9",
+    "content": "5790001330595"
+    },
+    "EnergyBusinessProcessRole": "D3M"
+    },
+    "TimeSeries": [
+    {
+    "TimeSeriesId": "74634301_86192545",
+    "OriginalMessageId": "bc8897bc5d5b4d8a9e7f72efe4b0d4c5",
+    "OriginalTimeSeriesId": "e1f06dee48d842c1a48b187065e710ff",
+    "EnergyTimeSeriesFunction": "9",
+    "EnergyTimeSeriesProduct": "8716867000030",
+    "EnergyTimeSeriesMeasureUnit": "KWH",
+    "TypeOfMP": "E17",
+    "SettlementMethod": "D01",
+    "AggregationCriteria": {
+    "MeteringPointId": "571051839308770693"
+    },
+    "Observation": [
+    {
+    "Position": 1,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 2,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 3,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 4,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 5,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 6,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 7,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 8,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 9,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 10,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 11,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 12,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 13,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 14,
+    "QuantityQuality": "E01",
+    "EnergyQuantity": 2.0
+    },
+    {
+    "Position": 15,
+    "QuantityQuality": "E01",
```

Comment:
```suggestion
    "QuantityQuality": "E01",
```

E01 is not a known `QuantityQuality` in this subsystem